ui2/src/pages/listing/Listing.tsx

```diff
@@ -0,0 +1,185 @@
+    if (!currencyAsset) return;
+    await ledger.exercise(ClearedMarketService.RequestFairValues, service.contractId, {
+      party,
+      currency: currencyAsset.payload.assetId,
+      optListingIds: selectedListingIds,
+    });
+  };
+  return (
+    <ModalFormErrorHandled onSubmit={() => requestFairValues()} title="Request FV">
+      <Form.Select
+        label="Currency"
+        placeholder="Select..."
+        required
+        options={assets.map(a => createDropdownProp(a.payload.assetId.label))}
+        value={currencyLabel}
+        onChange={(_, change) => setCurrencyLabel(change.value as string)}
+      />
+    </ModalFormErrorHandled>
+  );
+};
+
+const ListingComponent: React.FC<RouteComponentProps & ServicePageProps<Service>> = ({
+  services,
+}) => {
+  const party = useParty();
+  const { getName } = usePartyName(party);
+  const ledger = useLedger();
+  const { contractId } = useParams<any>();
+
+  const cid = contractId.replace('_', '#');
+
+  const { contracts: listings } = useStreamQueries(ListingTemplate);
+
+  const listing = listings.find(a => a.contractId === cid);
+  const service = services.find(s => s.payload.customer === party);
+
+  const { contracts: fairValueContracts, loading: fairValuesLoading } = useStreamQueries(FairValue);
+  const { contracts: clearedMarketServices } = useStreamQueries(ClearedMarketService);
+  const clearedMarketService = clearedMarketServices.find(
+    cms => cms.payload.customer === listing?.payload.customer
+  );
+  console.log(clearedMarketService);
+
+  const fairValues = fairValueContracts.filter(
+    fv => fv.payload.listingId === listing?.payload.listingId
+  );
+
+  const {
+    contracts: manualFVRequestContracts,
+    loading: manualFVRequestsLoading,
+  } = useStreamQueries(ManualFairValueCalculation);
+  const manualFVRequests = manualFVRequestContracts.filter(
+    fv => fv.payload.listingId === listing?.payload.listingId
+  );
+
+  const requestDisableDelisting = async () => {
+    if (!service || !listing) return; // TODO: Display error
+    await ledger.exercise(Service.RequestDisableListing, service.contractId, {
+      listingCid: listing.contractId,
+    });
+  };
+
+  return (
+    <div className="listing">
+      <Header as="h2">{listing?.payload.listingId}</Header>
+      <br />
+      <br />
+      <Tile header={<h4>Actions</h4>}>
+        <div className="action-row">
+          <Button.Group>
+            {!!clearedMarketService && (
+              <FairValueRequest
+                listingId={listing?.payload.listingId}
+                service={clearedMarketService}
+              />
+            )}
+            <Button floated="left" className="ghost" onClick={() => requestDisableDelisting()}>
+              Disable
+            </Button>
+          </Button.Group>
+        </div>
+      </Tile>
+      {!!listing && (
+        <Tile header={<h4>Details</h4>}>
+          <StripedTable
```

Comment:
u could add `loading` to line 73 and use it here to give this table a loading state